sematic/resolvers/cloud_resolver.py

```diff
@@ -324,6 +333,57 @@ def _process_run_output(self, run_id: str) -> None:
 
         self._update_future_with_value(future, value)
 
+    def _resolution_did_fail(
+        self, error: Exception, reason: Optional[str] = None
+    ) -> None:
+        if not isinstance(error, ResolverRestartError):
+            super()._resolution_did_fail(error)
+            logger.error("Really early return!")
+            return
+
+        try:
+            new_root_future = self._root_future.calculator(
+                **self._root_future.kwargs
+            ).set(name=self._root_future.props.name, tags=self._root_future.props.tags)
```

Comment:
This misses resource requirements.
Please introduce a copy or clone function on `Future`, and use that instead, in order to be future proof.